chall07/vde-dios.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+    "net/http"
+    "log"
+    "io/ioutil"
+    "fmt"
+    "regexp"
+    "strconv"
+    "strings"
+)
+
+func outputErr(err error){
+    if err != nil {
+        log.Fatalln(err)
+    }
+}
+
+func toHex(rgb [][]string) string {
+    output := ""
+    for i := 1; i < 4; i++ {
+        hex, err := strconv.Atoi(strings.Join(rgb[i],""))
+        outputErr(err)
+        output += fmt.Sprintf("%02x", hex)
+    }
+    return output
+}
+
+func MakeRequest(uri string) string{
+    resp, err := http.Get(uri)
+    outputErr(err)
+    body, err := ioutil.ReadAll(resp.Body)
+    outputErr(err)
+    defer resp.Body.Close()
```

Comment:
`defer` helps you run a function, no matter what happens. Here you want to close de response's body when the function exits, would it be because of a panic, an error, etc.  
Which means you should make the `defer` call earlier, for example here if `ioutil.ReadAll` fails for some reason, the body will never be closed
```suggestion
    resp, err := http.Get(uri)
    outputErr(err)
    defer resp.Body.Close()
    body, err := ioutil.ReadAll(resp.Body)
    outputErr(err)
```